java/build/bazel/tests/integration/WorkspaceDriver.java

```diff
@@ -270,4 +206,80 @@ private BazelWorkspaceDriverException(String message) {
       super(message);
     }
   }
+
+  /**
+   * bazel command with the given arguments. features like bazelrc file and environment can be added to result object
+   */
+  public BazelCommand bazelCommand(String arg, String... args){
+    return bazelCommand(Stream.concat(Stream.of(arg),Stream.of(args)).collect(Collectors.toList()));
+  }
+
+  public BazelCommand bazelCommand(List<String> args) {
+      if (currentBazel == null) {
+          throw new BazelWorkspaceDriverException("Cannot use bazel because no version was specified, "
+                  + "please call bazelVersion(version) before creating command(...).");
+      }
+      return new BazelCommand(args);
+  }
+
+  public class BazelCommand {
+
+      private final List<String> args;
+      private Optional<Path> bazelrcFile = Optional.empty();
+      private Map<String, String> environment = new HashMap<>();
+      private Path workingDirectory = Paths.get("");
+
+      private BazelCommand(List<String> args) {
+          this.args = Collections.unmodifiableList(args);
+      }
+
+      /**
+       * sets the working directory relative to workspace path
+       */
+      public BazelCommand withWorkingDirectory(Path relativeWorkingDirectory) {
```

Comment:
wdyt about using `inWorkingDirectory`?
Con: Less builder pattern
Pro: more fluent (bazelCommand(build //...).inWorkingDirectory(...)

Reply:
I like. WDYT about  `inSubDirectory`...? Also - why do we pass `Path` if it's relative to workspace root?